### ISSUE TITLE
Linux version doesn't build any more

### ISSUE THREAD

**Issue by SamuraiCrow** [khval/AmosKittens#34]
**Importent:**

Before posting bug repport, make sure you have the lastest retromode.library installed.
and have the lastest files from github.

**Describe the bug:**

In many places case sensitivity is not respected by the file system, eg. OS!=os directory, AmosKittens.h != amosKittens.h.  Makefile.linux is bitrotted as it doesn't pull in core_includes definitions.  sys_free needs to be free not FreeVec under Linux.

**To Reproduce:**

from BASH prompt:
make -fmakefile.linux

**Expected behavior**
(it compiles)

**Screenshots**

N/A

**Desktop (please complete the following information):**

 - OS: [Manjaro Linux]
 - OS Version [current]
 - CPU: [x86_64]
 - Hardware: [2nd generation i7 with 16 GB of RAM]
 
**Additional context**

N/A

**Comment by khval** [comment]
sys_free is macro is defined depending on the OS, Linux build has broken for some time now.

**Comment by khval** [comment]
Tokens has to be endian swapped before you can interpret the tokens.

**Comment by khval** [comment]
64bit not really supported, peek / poke / deek / doke, poke into real memory, we need to preallocate a block memory so a internal memory allocation routine from that block to avoid the 64bit memory space.

**Comment by SamuraiCrow** [comment]
Adding SDL2 as a dependency will allow assembly optimized endian swapping via sdl_endian.h if interested.

**Comment by khval** [comment]
Yes an Linux build need SDL, or Allegro, It might even be native X-Windows, what ever that API is.

**Comment by khval** [comment]
I guess a thread can be implemented using C++11 api, it does not matter really as we fallback on AmigaOS.

**Comment by SamuraiCrow** [comment]
SDL2 works on Windows, Mac, Linux and MorphOS if you want to combine trips.  Unlike version 1.2, SDL 2.0 wraps hardware graphics acceleration on all platforms.

**Comment by khval** [comment]
For Linux we need allow building extensions as .so objects, so that’s a lot work.

**Comment by khval** [comment]
retroModePC is also out date I guess.

**Comment by khval** [comment]
Sure SDL2 sounds ok, makes no difference it all graphic routines are CPU driven routines, its only when you have the final result and need to scale it has any use, no 3D graphics supported anyway.

**Comment by khval** [comment]
the Linux build will also need Menu support, currently we are using AmigaOS menus, we need fallback to legacy menus.

**Comment by SamuraiCrow** [comment]
We will also need to change the syntax of peek/poke/deek/doke/leek/loke to accept the bank start of the bank being read from or written to.  Otherwise sidestepping memory protection will result in a segmentation fault.  Any other direct memory accesses will be forbidden.  (This also solves the 64-bit support issues.) 
 
  On Fri, Apr 10, 2020 at 11:11 AM, Kjetil Hvalstrand<notifications@github.com> wrote:   


the Linux build will also need Menu support, currently we are using AmigaOS menus, we need fallback to legacy menus.

**Comment by khval** [comment]
Peek, poke, leek, loke also needs to work with local variables, 
Amos Basic “varptr” returns address of local variable.

**Comment by SamuraiCrow** [comment]
We'll just have to replace those special cases with specialized commands and functions like Var Peek() or Hot Spot X=varname.

**Comment by khval** [comment]
Yes sure it make sense provide commands make easier to code, but we also have some legacy to support.

**Comment by khval** [comment]
There is also the width of bob, that poke after, this of course gives really poor accuracy as its in number int16’s, and not pixels on classic Amos, we have more precise width in Amos Kittens.

**Comment by SamuraiCrow** [comment]
Bob banks and icon banks should be stored as texture coordinates, width and height on a huge texture so we can use the texture mapper on the graphics card to blit faster.

**Comment by khval** [comment]
Sure, we can do that, maybe hybridize retroMode fuse it with GL stuff.

**Comment by SamuraiCrow** [comment]
That's what SDL version 2 does, but better!  Windows uses DirectX, Mac uses Metal, Linux uses Mesa and MorphOS uses TinyGL.

**Comment by khval** [comment]
Well lets see what we can use or not the ARGB stuff is not defined yet anyway, so we do what ever like, but logical that use same commands on ARGB as you can on 8bit, the only thing know I wont to do, I forget the palette in ARGB format, as just extra useless lookup.
